### ISSUE TITLE
ISO: Do diff with ISO Word CSS

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/mn-native-pdf#20]
(This ticket is not ready to be acted on)

@opoudjis will be providing instructions to retrieve ISO Word CSS so that the PDF output can be 100% matching.

We will also need to do a visual diff between the two types of output.

**Comment by opoudjis** [comment]
Visual diff. Oh dear.

Pls find attached  CSS for the Word documents for ISO. They are cleaned up, and the corresponding Metanorma tags are given in comments after each style command.

[Αρχειοθήκη.zip](https://github.com/metanorma/mn-native-pdf/files/3914099/default.zip)

**Comment by ronaldtse** [comment]
For now I think the solution is this. We can have XSL-FO extract common values from CSS (e.g. some representation of CSS, YAML, XML, whatever).

https://stackoverflow.com/questions/17495152/using-external-css-in-xsl-fo/17548638

**Comment by Intelligent2013** [comment]
@ronaldtse 
We can't use CSS for 100% matching between Word and PDF.
The reasons:
- CSS contains msoffice specific styles like mso-padding-alt, mso-layout-grid-align, mso-border-alt, mso-level-number-format, mso-para-margin.
- CSS contains styles, which don't have analog in xsl-fo. For example: tab-stops. To emulate it in XSL-FO, it's necessary to use specific constructions.

When I wrote xslt, sure, I used the properties from CSS to maximal matching.
Regarding using of `<xsl:attribute-set name=...`, I think it's a good idea for define styles in a separate xslt.
But, on my opinion, is's good for small xslt, and it is more complex for maintenance of big code, because in case of change some style we need:
1. find element for which we want change style
2. goto to <xsl:attribute-set section and change style
3. check another element which uses these attribute sets to prevent undesirable changes.

In case of 'inline' style (as made) we only make change in local place of code.

**Comment by ronaldtse** [comment]
@Intelligent2013 are there any parameters that the CSS and XSL-FO stylesheets can share? The synchronization of both for a single type of document can be difficult without that.

**Comment by Intelligent2013** [comment]
@ronaldtse 
> @Intelligent2013 are there any parameters that the CSS and XSL-FO stylesheets can share? The synchronization of both for a single type of document can be difficult without that.
Yes - for example. font-family, font-size, font-style, font-weight, margin-*, text-align, text-decoration. They have identical behaviour.

**Comment by Intelligent2013** [comment]
But it's not prevent for such case, when paragraph use CSS class zzContents and redefine property 'margin-top':
```
<p class="zzContents" style="margin-top:0cm">
  <span lang="EN-GB" xml:lang="EN-GB">Contents</span>
</p>
```

```
p.zzContents, li.zzContents, div.zzContents {
  mso-style-name: zzContents;
  mso-style-noshow: yes;
  mso-style-unhide: no;
  mso-style-next: "TOC 1";
  margin-top: 48.0pt;
  margin-right: 0cm;

```
